log/ratelimit.go

```diff
@@ -23,106 +24,53 @@ type RateLimitedLogger struct {
 	discardedErrorLogLinesCounter prometheus.Counter
 }
 
-// NewRateLimitedLogger returns a logger.Interface that is limited to the given number of logs per second,
+// NewRateLimitedLogger returns a log.Logger that is limited to the given number of logs per second,
 // with the given burst size.
-func NewRateLimitedLogger(logger Interface, logsPerSecond rate.Limit, burstSize int, reg prometheus.Registerer) Interface {
```

Comment:
The main reason for this change was the introduction of `NewGoKitWithFields`, because with the `RateLimitedLoggerConfig` struct it was easier to pass the parameters.
But since we are going to remove `NewGoKitWithFields`, I will revert this change.